Pytest does not pass locally for a test_load_nifti test

### Environment details

- [x] I followed the [installation instructions](https://shimming-toolbox.org/en/latest/2_getting_started/1_installation.html)

Project version: 28d28b2bfc945383bf107c98209ab02f8e89b765

Operating System and version: MacOS Catalina 10.15.6

Python version: 3.7.3
Path to Python binary: /Users/alex/Documents/project/shimming-toolbox/venv/bin/python

### Bug Description
Running pytest locally throws an error for `test_load_nifti_multiple_echoes`.

<details>
 <summary>Summary Goes Here</summary>
 
```
(venv) alex@Alexandres-MacBook-Pro shimming-toolbox % pytest
=========================================================================================================== test session starts ===========================================================================================================
platform darwin -- Python 3.7.3, pytest-4.6.11, py-1.9.0, pluggy-0.13.1
rootdir: /Users/alex/Documents/School/Polytechnique/Master/project/shimming-toolbox
plugins: cov-2.5.1
collected 55 items                                                                                                                                                                                                                        

test/test_biot_savart.py ...                                                                                                                                                                                                        [  5%]
test/test_dicom_to_nifti.py .                                                                                                                                                                                                       [  7%]
test/test_example_scripts.py .                                                                                                                                                                                                      [  9%]
test/test_init.py .                                                                                                                                                                                                                 [ 10%]
test/test_load_nifti.py .....F....                                                                                                                                                                                                  [ 29%]
test/test_masking.py ......                                                                                                                                                                                                         [ 40%]
test/test_prelude.py ........                                                                                                                                                                                                       [ 54%]
test/test_sequential_zslice.py .                                                                                                                                                                                                    [ 56%]
test/test_siemens_basis.py .                                                                                                                                                                                                        [ 58%]
test/test_spherical_harmonics.py ...                                                                                                                                                                                                [ 63%]
test/simulate/test_numerical_model.py ....................                                                                                                                                                                          [100%]

================================================================================================================ FAILURES =================================================================================================================
________________________________________________________________________________________________ TestCore.test_load_nifti_multiple_echoes _________________________________________________________________________________________________

self = <test_load_nifti.TestCore object at 0x11f2f4860>, monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x11f2f4fd0>

    def test_load_nifti_multiple_echoes(self, monkeypatch):
        """
        Assert passes with correct data for multiple echoes
        :return:
        """
        if self.tmp_path.exists():
            shutil.rmtree(self.data_path_volume)
        monkeypatch.setattr('sys.stdin', StringIO('0\n'))
        niftis, info, json_info = load_nifti(self.tmp_path)
        assert (len(info) == 2), "Wrong number od info data 1"
        assert (len(json_info) == 2), "Wrong number of JSON data 1"
        self._json_phase['EchoNumber'] = 2
>       assert (json.dumps(json_info[0], sort_keys=True) == json.dumps(self._json_phase, sort_keys=True)), \
            "JSON file is not correctly loaded for first JSON1"
E       AssertionError: JSON file is not correctly loaded for first JSON1
E       assert '{"Acquisitio...mp": 222.944}' == '{"Acquisition...mp": 222.944}'
E         Skipping 357 identical leading characters in diff, use -v to show
E         Skipping 1213 identical trailing characters in diff, use -v to show
E         - oNumber": 1, "EchoTi
E         ?           ^
E         + oNumber": 2, "EchoTi
E         ?           ^

test/test_load_nifti.py:267: AssertionError
---------------------------------------------------------------------------------------------------------- Captured stdout call -----------------------------------------------------------------------------------------------------------
0:test_data

1:test_data_2

Enter the number for the appropriate acquisition folder, (type 'q' to quit) : 
================================================================================================== 1 failed, 54 passed in 24.77 seconds ===================================================================================================
```

</details>

### Reproduction Steps
Simply run `pytest` from the terminal when in shimming-toolbox directory

### Potential Reasons behind Bug:
I have not looked into it

Here's what is returned on Travis with #154 :

<details>
 <summary>Failure on travis</summary>
 
```
=================================== FAILURES ===================================
___________________ TestCore.test_load_nifti_multiple_echoes ___________________
self = <test_load_nifti.TestCore object at 0x7f9294477ed0>
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f92938ce750>
    def test_load_nifti_multiple_echoes(self, monkeypatch):
        """
        Assert passes with correct data for multiple echoes
        :return:
        """
        if self.tmp_path.exists():
            shutil.rmtree(self.data_path_volume)
        monkeypatch.setattr('sys.stdin', StringIO('0\n'))
        niftis, info, json_info = load_nifti(self.tmp_path)
        assert (len(info) == 2), "Wrong number od info data 1"
        assert (len(json_info) == 2), "Wrong number of JSON data 1"
        self._json_phase['EchoNumber'] = 1
>       assert (json.dumps(json_info[0], sort_keys=True) == json.dumps(self._json_phase, sort_keys=True)), \
            "JSON file is not correctly loaded for first JSON1"
E       AssertionError: JSON file is not correctly loaded for first JSON1
E       assert '{"Acquisitio...mp": 222.944}' == '{"Acquisition...mp": 222.944}'
E         - {"AcquisitionMatrixPE": 76, "AcquisitionNumber": 1, "AcquisitionTime": "16:21:2.480000", "BaseResolution": 128, "BodyPartExamined": "BRAIN", "ConsistencyInfo": "N4_VE11C_LATEST_20160120", "ConversionSoftware": "dcm2niix", "ConversionSoftwareVersion": "v1.0.20181125  (JP2:OpenJPEG) GCC9.3.0", "Dcm2bidsVersion": "2.1.4", "DeviceSerialNumber": "167006", "EchoNumber": 2, "EchoTime": 0.0025, "EchoTrainLength": 6, "FlipAngle": 25, "ImageComments": "phase", "ImageOrientationPatientDICOM": [1, 0, 0, 0, 1, 0], "ImageType": ["ORIGINAL", "PRIMARY", "M", "ND"], "ImagingFrequency": 123.259, "InPlanePhaseEncodingDirectionDICOM": "COL", "InstitutionAddress": "", "InstitutionName": "IUGM", "MRAcquisitionType": "2D", "MagneticFieldStrength": 3, "Manufacturer": "Siemens", "ManufacturersModelName": "Prisma_fit", "Modality": "MR", "PartialFourier": 1, "PatientPosition": "HFS", "PercentPhaseFOV": 59.375, "PhaseEncodingDirection": "j-", "PhaseEncodingSteps": 76, "PhaseResolution": 1, "PixelBandwidth": 600, "ProcedureStepDescription": "dev_acdc", "ProtocolName": "a_gre_DYNshim", "PulseSequenceDetails": "%CustomerSeq%_a_gre_DYNshim", "ReceiveCoilActiveElements": "HC7;NC1,2", "ReceiveCoilName": "HeadNeck_64", "ReconMatrixPE": 76, "RepetitionTime": 0.5, "SAR": 0.00453667, "ScanningSequence": "GR", "SequenceName": "fl2d6", "SequenceVariant": "SP", "SeriesDescription": "a_gre_DYNshim", "SeriesNumber": 6, "ShimSetting": [620, -7194, -9127, 77, 28, -20, -10, -23], "SliceThickness": 3, "SoftwareVersions": "syngo_MR_E11", "SpacingBetweenSlices": 3, "StationName": "MRC35049", "TxRefAmp": 222.944}
E         ?                                                                                                                                                                                                                                                                                                                                                                                ^^
E         + {"AcquisitionMatrixPE": 76, "AcquisitionNumber": 1, "AcquisitionTime": "16:21:2.480000", "BaseResolution": 128, "BodyPartExamined": "BRAIN", "ConsistencyInfo": "N4_VE11C_LATEST_20160120", "ConversionSoftware": "dcm2niix", "ConversionSoftwareVersion": "v1.0.20181125  (JP2:OpenJPEG) GCC9.3.0", "Dcm2bidsVersion": "2.1.4", "DeviceSerialNumber": "167006", "EchoNumber": 1, "EchoTime": 0.0025, "EchoTrainLength": 6, "FlipAngle": 25, "ImageComments": "phase", "ImageOrientationPatientDICOM": [1, 0, 0, 0, 1, 0], "ImageType": ["ORIGINAL", "PRIMARY", "M", "ND"], "ImagingFrequency": 123.259, "InPlanePhaseEncodingDirectionDICOM": "COL", "InstitutionAddress": "", "InstitutionName": "IUGM", "MRAcquisitionType": "2D", "MagneticFieldStrength": 3, "Manufacturer": "Siemens", "ManufacturersModelName": "Prisma_fit", "Modality": "MR", "PartialFourier": 1, "PatientPosition": "HFS", "PercentPhaseFOV": 59.375, "PhaseEncodingDirection": "j-", "PhaseEncodingSteps": 76, "PhaseResolution": 1, "PixelBandwidth": 600, "ProcedureStepDescription": "dev_acdc", "ProtocolName": "a_gre_DYNshim", "PulseSequenceDetails": "%CustomerSeq%_a_gre_DYNshim", "ReceiveCoilActiveElements": "HC7;NC1,2", "ReceiveCoilName": "HeadNeck_64", "ReconMatrixPE": 76, "RepetitionTime": 0.5, "SAR": 0.00453667, "ScanningSequence": "GR", "SequenceName": "fl2d6", "SequenceVariant": "SP", "SeriesDescription": "a_gre_DYNshim", "SeriesNumber": 6, "ShimSetting": [620, -7194, -9127, 77, 28, -20, -10, -23], "SliceThickness": 3, "SoftwareVersions": "syngo_MR_E11", "SpacingBetweenSlices": 3, "StationName": "MRC35049", "TxRefAmp": 222.944}
E         ?                                                                                                                                                                                                                                                                                                                                                                                ^^
test/test_load_nifti.py:267: AssertionError
```
</details>

--- a/shimmingtoolbox/load_nifti.py
+++ b/shimmingtoolbox/load_nifti.py
@@ -58,6 +58,7 @@ def load_nifti(path_data, modality='phase'):
     # Generate file_list
     file_list = []
     [file_list.append(os.path.join(path_data, f)) for f in os.listdir(path_data) if f not in file_list]
+    file_list = sorted(file_list)
 
     nifti_path = ""
     # Check for incompatible acquisition source path

--- a/shimmingtoolbox/load_nifti.py
+++ b/shimmingtoolbox/load_nifti.py
@@ -93,6 +93,7 @@ def load_nifti(path_data, modality='phase'):
 
     # Get a list of nii files
     nifti_list = [os.path.join(nifti_path, f) for f in os.listdir(nifti_path) if f.endswith((".nii", ".nii.gz"))]
+    nifti_list = sorted(nifti_list)
 
     # Read all images and headers available and store them
     nifti_init = [read_nii(nifti_list[i]) for i in range(len(nifti_list))]

--- a/test/test_load_nifti.py
+++ b/test/test_load_nifti.py
@@ -266,10 +266,10 @@ class TestCore(object):
         niftis, info, json_info = load_nifti(self.tmp_path)
         assert (len(info) == 2), "Wrong number od info data 1"
         assert (len(json_info) == 2), "Wrong number of JSON data 1"
-        self._json_phase['EchoNumber'] = 2
+        self._json_phase['EchoNumber'] = 1
         assert (json.dumps(json_info[0], sort_keys=True) == json.dumps(self._json_phase, sort_keys=True)), \
             "JSON file is not correctly loaded for first JSON1"
-        self._json_phase['EchoNumber'] = 1
+        self._json_phase['EchoNumber'] = 2
         assert (json.dumps(json_info[1], sort_keys=True) == json.dumps(self._json_phase, sort_keys=True)), \
             "JSON file is not correctly loaded for second JSON 1"
         assert (niftis.shape == (3, 3, 3, 2, 1)), "Wrong shape for the Nifti output data 1"
@@ -278,10 +278,10 @@ class TestCore(object):
         niftis, info, json_info = load_nifti(self.tmp_path)
         assert (len(info) == 2), "Wrong number of info data 2"
         assert (len(json_info) == 2), "Wrong number of JSON data 2"
-        self._json_phase['EchoNumber'] = 2
+        self._json_phase['EchoNumber'] = 1
         assert (json.dumps(json_info[0], sort_keys=True) == json.dumps(self._json_phase, sort_keys=True)), \
             "JSON file is not correctly loaded for first JSON 2"
-        self._json_phase['EchoNumber'] = 1
+        self._json_phase['EchoNumber'] = 2
         assert (json.dumps(json_info[1], sort_keys=True) == json.dumps(self._json_phase, sort_keys=True)), \
             "JSON file is not correctly loaded for second JSON 2"
         assert (niftis.shape == (3, 3, 3, 2, 1)), "Wrong shape for the Nifti output data 2"

